src/origin-matcher.js

```diff
@@ -1,4 +1,8 @@
 
+exports.generic = function (list) {
+  return list.length === 1 && list[0] === '*'
```

Comment:
Is there a case where we would want to explicitly match some domains, and wildcard the rest? Would it be surprising that `['*']` would have one behaviour for a domain matched against the wildcard and an entirely different behaviour for a domain matched against the wildcard in this list: `['foo.bar', '*']`